clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -326,6 +340,7 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
             SelectionKey key = iterator.next();
             iterator.remove();
             KafkaChannel channel = channel(key);
+            long channelStartTimeNanos = recordTimePerConnection ? time.nanoseconds() : 0;
```

Comment:
Our system test currently doesn't do perf validation well. It would be useful to just run ProducerPerformance and ConsumerPerformance and see if there is any noticeable degradation.